jupyter_ydoc/ydoc.py

```diff
@@ -10,87 +10,185 @@
+        :rtype: Any
+        """
         return self.get()
 
     @source.setter
-    def source(self, value):
+    def source(self, value: Any):
+        """
+        Sets the content of the document.
+
+        :param value: The content of the document.
+        :type value: Any
+        """
         return self.set(value)
 
     @property
-    def dirty(self) -> None:
+    def dirty(self) -> Optional[bool]:
+        """
+        Returns whether the document in memory has changes that are not on disk.
+
+        :return: Whether the document in memory has changes that are not on disk.
+        :rtype: bool | None
+        """
         return self._ystate["dirty"]
 
     @dirty.setter
     def dirty(self, value: bool) -> None:
+        """
+        Sets whether the document in memory has changes that are not on disk.
+
+        :param value: Whether the document in memory has changes that are not on disk.
+        :type value: bool
+        """
         with self._ydoc.begin_transaction() as t:
             self._ystate.set(t, "dirty", value)
 
     @property
-    def path(self) -> None:
+    def path(self) -> Optional[str]:
+        """
+        Returns document's path.
+
+        :return: Document's path.
+        :rtype: str | None
+        """
         return self._ystate.get("path")
 
     @path.setter
     def path(self, value: str) -> None:
+        """
+        Sets document's path.
+
+        :param value: Document's path.
+        :type value: str
+        """
         with self._ydoc.begin_transaction() as t:
             self._ystate.set(t, "path", value)
 
     @abstractmethod
-    def get(self):
+    def get(self) -> Any:
+        """
+        Returns the content of the document.
+
+        :return: Document's content.
+        :rtype: Any
+        """
         pass
 
     @abstractmethod
-    def set(self, value):
+    def set(self, value: Any) -> None:
+        """
+        Sets the content of the document.
+
+        :param value: The content of the document.
+        :type value: Any
+        """
         pass
 
     @abstractmethod
-    def observe(self, callback):
+    def observe(self, callback: Callable[[Any], None]) -> None:
+        """
+        Subscribes to document changes.
+
+        :param callback: Callback that will be called when the document changes.
+        :type callback: Callable[[Any], None]
+        """
         pass
 
-    def unobserve(self):
+    def unobserve(self) -> None:
+        """
+        Unsubscribes to document changes.
+
+        This method removes all the callbacks.
+        """
         for k, v in self._subscriptions.items():
             k.unobserve(v)
         self._subscriptions = {}
 
 
 class YFile(YBaseDoc):
     """
-    YFile class
+    Extends :class:`YBaseDoc`, and represents a plain text document.
+
+    Schema:
+
+    .. code-block:: json
+
+        {
+            "state": YMap,
+            "source": YText
+        }
     """
 
     def __init__(self, *args, **kwargs):
```

Comment:
I'm also wondering if it's possible to change the signature of the constructor to:

```suggestion
    def __init__(self, ydoc: Y.YDoc):
```

And if we should do it since that's the signature of the base class and `ydoc` is a required parameter.

Reply:
The same for the YNotebook class

Reply:
Yes we could change it.